### PR TITLE
[Test][RayCluster] Test redis cleanup job in the e2e compatibility test

### PR DESCRIPTION
**Pull request by rueian**
## Why are these changes needed?

As discussed with @kevin85421 offline, we think that we should test the redis cleanup job in the `tests/compatibility-test.py`.

This PR appends the `cleanup_redis` procedure to every test in the `RayFTTestCase` to test the cleanup job by deleting the RayCluster CR and watching the Redis' `DBSIZE` to zero. If it doesn't go zero before timeout, then the test will fail.

## Related issue number

Fixes https://github.com/ray-project/kuberay/issues/1997

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: tests/compatibility-test.py [hunk 1]
```diff
@@ -86,8 +87,9 @@ class RayFTTestCase(unittest.TestCase):
     cluster_template = CONST.REPO_ROOT.joinpath("tests/config/ray-cluster.ray-ft.yaml.template")
     ray_cluster_ns = "default"
 
-    @classmethod
```
Comment (kevin85421) [inline tests/compatibility-test.py:89]: How about creating a new test for cleaning up Redis? That is, `test_ray_serve` -> `test_detached_actor` -> `test_cleanup_redis`.

**Review by kevin85421 — COMMENTED** [review]
We can also remove the Redis cleanup check in the sample YAML tests.

File: tests/compatibility-test.py (hunk 1 above)
Comment (rueian) [reply tests/compatibility-test.py:89]: That’s was my first attempt. Unfortunately, the `test_cleanup_redis` was going to be the first test case that run by the Python `unittest` framework and the test case was going to delete the RayCluster CR required by the rest of the cases.

The execution order of tests under a `unittest.TestCase` is alphabetical by their method names. Therefore, if we want to use a standalone case to test redis cleanup, we may need to encode sequence into all test names. For example: `test_01_ray_serve` -> `test_02_detached_actor` -> `test_ 99_cleanup_redis`

**Review by kevin85421 — APPROVED** [review]

**Comment by kevin85421** [comment]
The CI stuck for 10 hours.

Screen Shot 2024-03-20 at 3 15 32 PM
Screen Shot 2024-03-20 at 3 16 56 PM

**Comment by rueian** [comment]
> The CI stuck for 10 hours. I cancel it.
> 
> Screen Shot 2024-03-20 at 3 15 32 PM Screen Shot 2024-03-20 at 3 16 56 PM

I will look into why did it stuck.

**Review by kevin85421 — APPROVED** [review]

**Pull request merged**